internal/jobservice/repository/sqlite.go

```diff
@@ -374,3 +367,31 @@ func (s *JSRepoSQLite) GetSubscribedJobSets(ctx context.Context) ([]SubscribedTu
 	}
 	return tuples, nil
 }
+
+func (s *JSRepoSQLite) purgeExpired() {
+	jobsetsStmt := fmt.Sprintf(`DELETE FROM jobsets WHERE Timestamp < (UNIXEPOCH() - %d);`, s.jobServiceConfig.PurgeJobSetTime)
+	jobServiceStmt := fmt.Sprintf(`DELETE FROM jobservice WHERE Timestamp < (UNIXEPOCH() - %d);`, s.jobServiceConfig.PurgeJobSetTime)
+
+	ticker := time.NewTicker(time.Duration(s.jobServiceConfig.PurgeJobSetTime) * time.Second)
+	log := log.WithField("JobService", "ExpiredJobSetsPurge")
+
+	purge := func(table, sqlStmt string) {
+		result, err := s.db.Exec(jobsetsStmt)
+		if err != nil {
+			log.Errorf("error deleting expired records from table %s: %v ", table, err)
+		} else {
+			count, err := result.RowsAffected()
+			if err != nil {
+				log.Errorf("error getting affected rows for expired %s records deletion: %v", table, err)
+			}
+			log.Debugf("Deleted %d expired records in table %s", count, table)
+		}
+	}
+
+	for range ticker.C {
```

Comment:
I would also say it would be useful to explicitly log that the purge is starting once the loop starts.

Reply:
> Is there a reason that `purgeExpired` has two different implementations, one for sqlite and one for postgres? Naively I'd think that a better implementation would be to have a single `purgeExpired` function that can compose the cleanup operations for each kind of db backend.

I find it hard to think how we could have a single `purgeExpired` implementation due to differences in each db handler interface. `JSRepoSQLite` uses `database/sql.DB` as a db handler while `JSRepoPostgres` uses `pgxpool.Pool`. 

However, I believe we can add `purgeExpired` to the `SQLJobService` interface and expose its implementations so it can be launched from `Startup` instead of `Setup` as you suggested below